app/models/concerns/progress.rb

```diff
@@ -0,0 +1,17 @@
+module Progress
+  extend ActiveSupport::Concern
+
+  included do
+    before_save :parent, unless: :parent_id?
+    belongs_to :parent, class_name: 'Indicator', optional: true
+  end
+
+  def parent
+    super || assign_attributes(parent: parent_content&.progress_for(user, organization))
+    super
```

Comment:
Using one-liner ifs is more intention revealing, since the or-expression value is not important: 

```ruby
assign_attributes(parent: parent_content&.progress_for(user, organization)) unless super
```